tests/settings/ConfigurationGatheringTest.py

```diff
@@ -83,13 +84,39 @@ def test_gather_configuration(self):
 
         self.assertEqual(len(local_bears['cli']), 0)
 
-    def test_default_coafile_parsing(self):
+    def test_default_coafile_deprecation(self):
+        '''
+        This test is temporary, it will send a directory
+        path which doesnot contain system_coafile named file
+        to function in Constants.py and ask for allocating
+        a system_coafile. Since there is no file nameds
+        system_coafile, it should log a warning
+        '''
+
+        tmp = Constants.system_coafile
+        logger = logging.getLogger()
+
+        system_coafile_path = os.path.abspath(os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            'section_manager_test_files',
+            'child_dir'))
+
+        with self.assertLogs(logger, 'WARNING') as cm:
+            get_system_coafile(system_coafile_path)
```

Comment:
you can use `LogCapture` if you like, then you don't have to retrieve the logger with `getLogger()` ;)

Reply:
together with `LogCapture.check` which works imo a bit simpler than asserting for `cm.output` ;)